File: src/main/java/com/SSweb/blog/pojo/Article.java
package com.SSweb.blog.pojo;


import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import org.hibernate.validator.constraints.URL;

import java.time.LocalDateTime;

@Data
public class Article {

    private Integer id;

    @NotEmpty
    @Pattern(regexp = "^\\S{1,10}$")
    private String title;
    private String content;

    @URL
    private String coverImg;
    private String state;
    private Integer categoryId;
    private Integer createUser;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;
}

File: src/assets/script/pojo/article.js
export class Article {
    id;
    title;
    content;
    coverImg;
    state;
    categoryId;
    createUser;
    createTime;

    constructor(id, title, content, coverImg, state, categoryId, createUser, createTime) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.coverImg = coverImg;
        this.state = state;
        this.categoryId = categoryId;
        this.createUser = createUser;
        this.createTime = createTime;
    }
}

export class PreviewArticle {
    id;
    title;
    coverImg;
    category;
    categoryId;
    createUser;
    createTime;

    constructor(id, title, coverImg, categoryId, createUser, createTime) {
        this.id = id;
        this.title = title;
        this.coverImg = coverImg;
        this.categoryId = categoryId;
        this.createUser = createUser;
        this.createTime = createTime;
    }
}